### PR TITLE
cephadm: infer fsid from ceph.conf

### PR DESCRIPTION
**Pull request by mgfritch**
Fixes: https://tracker.ceph.com/issues/51328
Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -4455,24 +4461,15 @@ def command_run(ctx):
...
 
-def fsid_conf_mismatch(ctx):
-    # type: (CephadmContext) -> bool
-    (config, _) = get_config_and_keyring(ctx)
-    if config:
-        for c in config.split('\n'):
-            if 'fsid = ' in c.strip():
-                if 'fsid = ' + ctx.fsid != c.strip():
-                    return True
-    return False
```
Comment (sebastian-philipp) [inline src/cephadm/cephadm:4466]: this failed, cause there was no daemon on that host, right?

Comment (mgfritch) [reply src/cephadm/cephadm:4466]: this block was refactored into `command_shell` for simplicity.

the failure occurred during `infer_fsid`:
https://github.com/ceph/ceph/blob/a408347e148432c8a8264b0a0022d49f096fc587/src/cephadm/cephadm#L1657-L1668

**Review by sebastian-philipp — APPROVED** [review]

**Review by tchaikov — DISMISSED** [review]
```
2021-07-01T15:35:17.552 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]: cephadm 2021-07-01T15:35:17.235237+0000 mgr.y (mgr.14150) 509 : cephadm [ERR] cephadm exited with an error code: 1, stderr:Traceback (most recent call last):
2021-07-01T15:35:17.552 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:   File "/var/lib/ceph/a5dedcd4-da7f-11eb-8c1c-001a4aab830c/cephadm.34ef8889a625e5afa2b96d889ff63e7ca9cf27b03a27a59bbf31a29d20596389", line 8283, in <module>
2021-07-01T15:35:17.552 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:     main()
2021-07-01T15:35:17.553 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:   File "/var/lib/ceph/a5dedcd4-da7f-11eb-8c1c-001a4aab830c/cephadm.34ef8889a625e5afa2b96d889ff63e7ca9cf27b03a27a59bbf31a29d20596389", line 8271, in main
2021-07-01T15:35:17.553 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:     r = ctx.func(ctx)
2021-07-01T15:35:17.553 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:   File "/var/lib/ceph/a5dedcd4-da7f-11eb-8c1c-001a4aab830c/cephadm.34ef8889a625e5afa2b96d889ff63e7ca9cf27b03a27a59bbf31a29d20596389", line 1700, in _infer_config
2021-07-01T15:35:17.553 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:     name = ctx.name
2021-07-01T15:35:17.553 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:   File "/var/lib/ceph/a5dedcd4-da7f-11eb-8c1c-001a4aab830c/cephadm.34ef8889a625e5afa2b96d889ff63e7ca9cf27b03a27a59bbf31a29d20596389", line 155, in __getattr__
2021-07-01T15:35:17.554 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]:     return super().__getattribute__(name)
2021-07-01T15:35:17.554 INFO:journalctl@ceph.mon.a.smithi123.stdout:Jul 01 15:35:17 smithi123 bash[11468]: AttributeError: 'CephadmContext' object has no attribute 'name'
```

see https://pulpito.ceph.com/kchai-2021-07-01_14:09:56-rados-wip-kefu-testing-2021-07-01-1912-distro-basic-smithi/6247309/

for more instances, see https://pulpito.ceph.com/kchai-2021-07-01_14:09:56-rados-wip-kefu-testing-2021-07-01-1912-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-07-20_13:35:25-rados:cephadm-wip-swagner-testing-2021-07-20-1243-distro-basic-smithi/

**Pull request merged**